include/flatbuffers/flatbuffers.h

```diff
@@ -697,6 +697,24 @@ class vector_downward {
     }
   }
 
+  void reset(size_t initial_size = 1024, Allocator *allocator = nullptr,
+             bool own_allocator = false) {
```

Comment:
Okay I took a look at this and it was indeed an artifact of some older code. It wasn't performing a dynamic allocation, but it was setting some allocator pointers that don't need to be set (they don't change).

I've updated this to just be a plain `reset()` func that deallocates the old buffer and reallocates a new one with a capacity based on the original `initial_size`.